src/serverlist.cpp

```diff
@@ -490,6 +541,105 @@ void CServerListManager::CentralServerQueryServerList ( const CHostAddress& Inet
     }
 }
 
+int CServerListManager::IndexOf ( CHostAddress haSearchTerm )
+{
+    // Find the server in the list. The very first list entry
+    // per definition is the directory server
+    // (i.e., this server).
+    for ( int iIdx = ServerList.size() - 1; iIdx > 0; iIdx-- )
+    {
+        if ( ServerList[iIdx].HostAddr == haSearchTerm )
+        {
+            return iIdx;
+        }
+    }
+    return INVALID_INDEX;
+}
+
+void CServerListManager::CentralServerLoadServerList ( const QString strServerList )
+{
+    QFile file ( strServerList );
+
+    if ( !file.open ( QIODevice::ReadWrite | QIODevice::Text ) )
+    {
+        qWarning() << qUtf8Printable ( QString ( "Could not open '%1' for read/write" ).arg ( ServerListFileName ) );
+        return;
+    }
+
+    // use entire file content for the persistent server list
+    // and remember it for later writing
+    qInfo() << "Setting persistent server list file:" << strServerList;
+    ServerListFileName = strServerList;
+    CHostAddress haServerHostAddr;
+
+    QTextStream in ( &file );
+    while ( !in.atEnd() )
+    {
+        QString     line   = in.readLine();
+        QStringList slLine = line.split ( ";" );
+        if ( slLine.count() != 7 )
+        {
+            qWarning() << qUtf8Printable ( QString ( "Could not parse '%1' successfully - bad line" ).arg ( line ) );
```

Comment:
As above. Can the user do anything? Fix the file, look for syntax errors,...?

Reply:
Undo whatever manual edit they did to break it ;)

Reply:
So it should say that.

Reply:
Well, that's guessing and potentially wrong information.  I really, really don't like messages that try to guess what's wrong and suggest potentially incorrect things to do on the basis of too little information.  Either the user knows what they've done - and therefore knows without being told - or they don't and they can't do anything about it.  Putting annoying guidance in the message is simply that - annoying.

So no.